awscli/autocomplete/local/basic.py

```diff
@@ -439,10 +439,12 @@ def _get_prompt_for_list(self, arg_model, parsed_input):
         # we have two way we can enter lists:
         # - if it's a top level structure it will be space separated and
         # parsed_input will contain the last element of the list
-        # - if it's a nested structure we'll get the whole list as an input
-        if not isinstance(parsed_input, list):
-            parsed_input = [parsed_input]
-        return self._get_completion(arg_model.member, parsed_input[-1])
+        # - if it's a nested structure we'll get the whole list as an input 
+        # and take the last item
+        # - if list is empty we'll pass None to the next completer
+        if isinstance(parsed_input, list):
+            parsed_input = ([None] + parsed_input)[-1]
```

Comment:
This is still hard to follow even with the comments. Could we organize it to follow the items in the comment? So something like this:
```python
if isinstance(parsed_input, list):
     if parsed_input:
        parsed_input = parsed_input[-1]
     else:
        parsed_input = None
```